Tests/tests/test_MotionDetection.cpp

```diff
@@ -0,0 +1,309 @@
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getSensitivity")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getLastMotionEventElapsedTime")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setMotionEventsActivePeriod")));
+}
+
+TEST_F(MotionDetectionEventTest, getMotionDetectors)
+{
+		EXPECT_CALL(motionDetectionImplMock, MOTION_DETECTION_GetMotionDetectors(::testing::_))
+			.Times(1)
+			.WillOnce(::testing::Invoke(   
+				[](MOTION_DETECTION_CurrentSensorSettings_t *pSensorStatus) {
+                pSensorStatus = (MOTION_DETECTION_CurrentSensorSettings_t *)malloc(sizeof(MOTION_DETECTION_CurrentSensorSettings_t));
+                memset(pSensorStatus, 0, sizeof(MOTION_DETECTION_CurrentSensorSettings_t));
+
+                strcpy(pSensorStatus->m_sensorIndex, MOTION_DETECTOR);
+                strcpy(pSensorStatus->m_sensorDescription, MOTION_DETECTION_DESCRIPTION);
+                strcpy(pSensorStatus->m_sensorType, MOTION_DETECTOR_TYPE);
+                pSensorStatus->m_sensorDistance = MOTION_DETECTION_DISTANCE;
+                pSensorStatus->m_sensorAngle = MOTION_DETECTION_ANGLE;
+                pSensorStatus->m_sensitivityMode = 2;
+
+                strcpy(pSensorStatus->m_sensitivity[SENSITIVITY_IDENTIFIER_1], STR_SENSITIVITY_LOW);
+                strcpy(pSensorStatus->m_sensitivity[SENSITIVITY_IDENTIFIER_2], STR_SENSITIVITY_MEDIUM);
+                strcpy(pSensorStatus->m_sensitivity[SENSITIVITY_IDENTIFIER_3], STR_SENSITIVITY_HIGH);
+
+                return MOTION_DETECTION_RESULT_SUCCESS;
+            }));
+
+        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getMotionDetectors"), _T("{}"), response));
+        EXPECT_EQ(response,  string("{\"supportedMotionDetectors\":[\"\x1F\"],\"supportedMotionDetectorsInfo\":{\"\x1F\":{\"description\":\"\",\"type\":\"\",\"distance\":\"0\",\"angle\":\"0\",\"sensitivityMode\":\"4278187664\"}},\"success\":true}"));
+}
+
+TEST_F(MotionDetectionEventTest, armmotiondetected)
+{
+	EXPECT_CALL(motionDetectionImplMock, MOTION_DETECTION_ArmMotionDetector(::testing::_,::testing::_))
+    .Times(1)
+    .WillOnce(::testing::Invoke(
+		[](MOTION_DETECTION_Mode_t mode,std::string index) {
+                return MOTION_DETECTION_RESULT_SUCCESS;
+            }));
+
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("arm"), _T("{\"index\":\"FP_MD\",\"mode\":\"1\" }"), response));
+    EXPECT_EQ(response,  string("{\"success\":true}"));
+}
+
+TEST_F(MotionDetectionEventTest, armmotiondetectedInvalid)
+{
+    EXPECT_CALL(motionDetectionImplMock, MOTION_DETECTION_ArmMotionDetector(::testing::_,::testing::_))
+    .Times(1)
+    .WillOnce(::testing::Invoke(
+		[](MOTION_DETECTION_Mode_t mode, std::string index) {
+                return MOTION_DETECTION_RESULT_INDEX_ERROR;
+            }));
+
+    EXPECT_EQ(Core::ERROR_GENERAL, handler.Invoke(connection, _T("arm"), _T("{ \"index\":\"FP_MD\",\"mode\":\"10\"}"), response));
```

Comment:
please pass params same as in armmotiondetected.
the reader will not understand - does it return ERROR because of MOTION_DETECTION_RESULT_INDEX_ERROR, or because input mode is 10?
it is unnecessary confusion. Tests should be clear